client/components/ReviewsOverview.jsx

```diff
@@ -6,8 +6,8 @@ const ReviewsOverview = {
   reviewScores: [5, 4, 3, 2, 1],
 
   totalReviewsAverageRating(reviews) {
```

Comment:
@Matt-Wigg , you can make your server API more useful, and reduce frontend complexity by moving this functionality to the backend code (either as a computed field of the database query, or computed value of the server functionality). This way, you can reuse the computed value anywhere in your frontend code. 

**You should definitely challenge this argument by trying to answer the question: "Where is the best place to host this logic, and what are the possible pros and cons in that case?"**

Reply:
I love that you have mentioned this!!!! 

Here is my journal entry for the day: 
> At the moment, I am doing one GET request for all reviews and filtering them front-end. Is this best practice???? Or should I trigger a GET request anytime a user interacts with a filtering/show more event? What are the pros and cons of minimizing database queries?